### PR TITLE
Display serial number and battery level

### PR DESCRIPTION
**Pull request by neon-ninja**
Hi! Thanks for making this open source. This PR adds the device serial number and battery level / charging status to the navbar. Additionally, in order to have this update on each page load / refresh, new HTML is generated whenever index.html is requested.

Screenshot: 
image

### CODE REVIEW

File: data/src/main/kotlin/info/dvkr/screenstream/data/httpserver/HttpServerFiles.kt [hunk 1]
```diff
@@ -350,6 +351,15 @@ class HttpServerFiles(context: Context, private val settingsReadOnly: SettingsRe
...
         coplayObject.put("appsList", appsArray)
+
+        val batteryObject = JSONObject()
+        val bm: BatteryManager = this.applicationContext.getSystemService("batterymanager") as BatteryManager
+        batteryObject.put("level", bm.getIntProperty(BatteryManager.BATTERY_PROPERTY_CAPACITY))
+        batteryObject.put("charging", bm.isCharging())
+        coplayObject.put("battery", batteryObject)
+
+        coplayObject.put("serial", android.os.Build.SERIAL)
+
```
Comment (sparklightlabs) [inline data/src/main/kotlin/info/dvkr/screenstream/data/httpserver/HttpServerFiles.kt:384]: This is awesome! I went hunting for a way to add the serial number but every method I found was deprecated.

File: data/src/main/kotlin/info/dvkr/screenstream/data/httpserver/HttpServerRxHandler.kt [hunk 2]
```diff
@@ -106,8 +103,9 @@ internal class HttpServerRxHandler(
             uri == HttpServerFiles.FULLSCREEN_ON_PNG_ADDRESS -> response.sendPng(httpServerFiles.fullScreenOnPng)
             uri == HttpServerFiles.FULLSCREEN_OFF_PNG_ADDRESS -> response.sendPng(httpServerFiles.fullScreenOffPng)
             uri.startsWith(systemActionAddress) -> onSystemAction(uri.substringAfter("?")).run {response.empty() } // possible get rid of sending a response...
-            uri == HttpServerFiles.DEFAULT_HTML_ADDRESS -> response.sendHtml(if (pinEnabled) pinRequestHtml else indexHtml)
-            uri == pinAddress && pinEnabled -> response.sendHtml(indexHtml)
+            uri == HttpServerFiles.DEFAULT_HTML_ADDRESS -> response.sendHtml(if (pinEnabled) pinRequestHtml else
+                httpServerFiles.configureDirectorHTML(streamAddress, appActionAddress, systemActionAddress))
+            uri == pinAddress && pinEnabled -> response.sendHtml(httpServerFiles.configureDirectorHTML(streamAddress, appActionAddress, systemActionAddress))
```
Comment (sparklightlabs) [inline data/src/main/kotlin/info/dvkr/screenstream/data/httpserver/HttpServerRxHandler.kt:116]: Great idea. I need to go in and configure the webpage to trigger a refresh when settings like picture quality are changed. Right now it just kills the webpage and you have to refresh it manually from the browser.

File: data/src/main/kotlin/info/dvkr/screenstream/data/httpserver/HttpServerFiles.kt [hunk 3]
```diff
@@ -75,6 +75,7 @@ class HttpServerFiles(context: Context, private val settingsReadOnly: SettingsRe
         const val TOGGLE_STREAM_ADDRESS = "toggle-stream"
         const val CHANGE_IMAGE_SIZE_ADDRESS = "image-size"
         const val CHANGE_IMAGE_COMPRESSION_ADDRESS = "image-compression"
+        const val GET_JSON_ADDRESS = "/get-json"
```
Comment (sparklightlabs) [inline data/src/main/kotlin/info/dvkr/screenstream/data/httpserver/HttpServerFiles.kt:78]: We might need to make a change here. The way pin security is set up, it garbles the address for certain endpoints like the image stream and picture quality. We will want to be able to garble the json address as well. 

Let me look into this a little more.

File: data/src/main/kotlin/info/dvkr/screenstream/data/httpserver/HttpServerFiles.kt (hunk 1 above)
Comment (neon-ninja) [reply data/src/main/kotlin/info/dvkr/screenstream/data/httpserver/HttpServerFiles.kt:384]: Technically android.os.Build.SERIAL is deprecated, but it works on the Quest as the Quest is running SDK25 (https://stackoverflow.com/a/11029386)

File: data/src/main/kotlin/info/dvkr/screenstream/data/httpserver/HttpServerFiles.kt (hunk 3 above)
Comment (neon-ninja) [reply data/src/main/kotlin/info/dvkr/screenstream/data/httpserver/HttpServerFiles.kt:78]: This endpoint calls:

```
response.sendJSON(httpServerFiles.configureCoPlayJSON(
                        httpServerFiles.configureStreamAddress(),
                        httpServerFiles.configureAppActionAddress(),
                        httpServerFiles.configureSystemActionAddress()
                ))
```

And the logic for PIN garbling is contained within those 3 configure functions, so should still work

**Comment by neon-ninja** [comment]
Also FYI, I wrote a Python script to scan a network for devices running CoPlay to collect JSON data and store it in a MySQL database - https://github.com/UoA-eResearch/CoPlay-Collector/blob/master/collect.py

**Comment by sparklightlabs** [comment]
You are a rockstar, my friend. Thank you for your contributions. If you don't mind me asking, what's your use case for CoPlay?

**Comment by neon-ninja** [comment]
We have 33 Oculus Quests spread across 2 campuses at the University of Auckland, I need to be able to remotely monitor them. For example, when I develop a new version of a research application, I want to know what devices have what version installed, where its located, it's IP address and battery level, etc.

**Pull request merged**